xrpl/transaction/main.py

```diff
@@ -198,3 +230,67 @@ def transaction_json_to_binary_codec_form(dictionary: Dict[str, Any]) -> Dict[st
         A new dictionary object that has been reformatted.
     """
     return model_transaction_to_binary_codec(dictionary)
+
+
+def _check_fee(transaction: Transaction, client: Optional[Client] = None) -> None:
+    """Checks if the Transaction fee is lower than the expected Transaction type fee"""
+    # Calculate the expected fee from the network load and transaction type
+    expected_fee = _calculate_fee_per_transaction_type(transaction, client)
+
+    if transaction.fee and int(transaction.fee) > int(expected_fee):
+        raise XRPLException(
+            "Fee value: "
+            + str(drops_to_xrp(transaction.fee))
+            + " XRP exceeds the "
+            + str(drops_to_xrp(expected_fee))
+            + " maximum XRP fee limit for "
+            + transaction.transaction_type
+            + " transaction"
+        )
+
+
+def _calculate_fee_per_transaction_type(
+    transaction: Transaction, client: Optional[Client] = None
+) -> str:
+    """
+    Calculate the total fee in drops for a transaction based on:
+    - the network fee
+    - the transaction condition
+
+    https://xrpl.org/transaction-cost.html#special-transaction-costs
+
+    Args:
+        transaction: the Transaction to be submitted.
+        client: the network client with which to submit the transaction.
+
+    Returns:
+        The expected Transaction fee in drops
+    """
+    # Reference Transaction (Most transactions)
+    if not client:
+        net_fee = 10  # 10 drops
+    else:
+        net_fee = int(get_fee(client))  # Usually 0.00001 XRP (10 drops)
+
+    base_fee = net_fee
+
+    # EscrowFinish Transaction with Fulfillment
+    # https://xrpl.org/escrowfinish.html#escrowfinish-fields
+    if transaction.transaction_type == TransactionType.ESCROW_FINISH:
+        escrow_finish = cast(EscrowFinish, transaction)
+        if escrow_finish.fulfillment:
+            fulfillment_bytes = escrow_finish.fulfillment.encode("ascii")
+            # 10 drops × (33 + (Fulfillment size in bytes / 16))
+            base_fee = math.ceil(net_fee * (33 + (len(fulfillment_bytes) / 16)))
+
+    # AccountDelete Transaction
+    if transaction.transaction_type == TransactionType.ACCOUNT_DELETE:
+        base_fee = 5000000
```

Comment:
This should be a named constant instead of a buried magic number

Reply:
I created a constant for it